usaspending_api/etl/management/commands/update_file_c_linkages_in_delta.py

```diff
@@ -0,0 +1,159 @@
+            required=False,
+            help="Whether to create a full new `int` table instead of a clone.",
+        )
+
+        parser.add_argument(
+            "--spark-s3-bucket",
+            type=str,
+            required=False,
+            default=CONFIG.SPARK_S3_BUCKET,
+            help="The destination bucket in S3 to write the data",
+        )
+
+    def get_unlinked_count(self, schema):
+        unlinked_sql = f"""
+        SELECT
+            count(*)
+        FROM
+            {schema}.financial_accounts_by_awards
+        WHERE
+            award_id IS NULL;
+        """
+
+        results = self.spark.sql(unlinked_sql)
+        return results.collect()[0][0]
+
+    def handle(self, *args, **options):
+
+        extra_conf = {
+            # Config for Delta Lake tables and SQL. Need these to keep Dela table metadata in the metastore
+            "spark.sql.extensions": "io.delta.sql.DeltaSparkSessionExtension",
+            "spark.sql.catalog.spark_catalog": "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+            # See comment below about old date and time values cannot parsed without these
+            "spark.sql.legacy.parquet.datetimeRebaseModeInWrite": "LEGACY",  # for dates at/before 1900
+            "spark.sql.legacy.parquet.int96RebaseModeInWrite": "LEGACY",  # for timestamps at/before 1900
+            "spark.sql.jsonGenerator.ignoreNullFields": "false",  # keep nulls in our json
+        }
+
+        no_clone = options["no_clone"]
+        spark_s3_bucket = options["spark_s3_bucket"]
+
+        self.spark = get_active_spark_session()
+        spark_created_by_command = False
+        if not self.spark:
+            spark_created_by_command = True
+            self.spark = configure_spark_session(**extra_conf, spark_context=self.spark)  # type: SparkSession
+
+        # Setup Logger
+        logger = get_jvm_logger(self.spark, __name__)
+
+        # Setup int table. Creates a shallow clone of the `raw` FABA table in the `int` schema.
+        # If the --no-clone option is provided a full table is created instead.
+        if no_clone:
+            self.spark.sql("DROP TABLE IF EXISTS int.financial_accounts_by_awards")
+            call_command(
+                "create_delta_table",
+                "--destination-table",
+                "financial_accounts_by_awards",
+                "--spark-s3-bucket",
+                spark_s3_bucket,
+                "--alt-db",
+                "int",
+            )
+            self.spark.sql(
+                "INSERT OVERWRITE int.financial_accounts_by_awards SELECT * FROM raw.financial_accounts_by_awards;"
+            )
+        else:
+            self.spark.sql(
+                """
+            CREATE OR REPLACE TABLE int.financial_accounts_by_awards
+            SHALLOW CLONE raw.financial_accounts_by_awards;
+            """
+            )
+
+        # Log initial linkage count
+        unlinked_count = self.get_unlinked_count("raw")
```

Comment:
Is this right, that you want to do the initial count from `raw.faba` and not the newly cloned `int.faba`? Maybe it's the same since it just go cloned? Seems it would make more sense / be more readable if the before/after counts were on the same table from the same schema.

Also, what is the provenance of that count?
- at this point, raw.faba is copied up from postgres
- which hopefully had its postgres data c-to-d linked from the previous pipeline run before it was copied up
- I guess this all makes sense if there are no changes to faba in postgres from when it is copied up to delta, and when the PG linkage happens

Reply:
You're right that we technically shouldn't see any difference between running this on `raw` and `int`. I can see your point thought that getting the initial unlinked count from `int` might be more readable though. I'll update.